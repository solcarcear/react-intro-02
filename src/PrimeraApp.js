import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({saludo, subtitulo}) => {
    return (
        <>
            <h1>{saludo}</h1>
            <p>{subtitulo}</p>
        </>
    );
};
PrimeraApp.propTypes={
    saludo:PropTypes.string.isRequired
};
PrimeraApp.defaultProps={
    saludo:'Hola Mundo',
    subtitulo:'Esta es mi primera vez',
};

export default PrimeraApp;