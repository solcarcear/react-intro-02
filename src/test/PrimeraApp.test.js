import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import PrimeraApp from '../PrimeraApp';


describe('Pruebas en primera app', () => {
    // test('should return the message "Hola soy carlos" ', () => {
    //     const saludo = 'Hola soy carlos';
    //     const { getByText } = render(<PrimeraApp saludo={saludo} />);
    //     expect(getByText(saludo)).toBeInTheDocument();
    // });

    test('should return <PrimeraApp/> correctamente', () => {
        const saludo = 'Hola soy carlos';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('should return subt text', () => {
        const saludo = 'Hola soy carlos';
        const subTitulo = 'Hola soy un subitulo';
        const wrapper = shallow(
            <PrimeraApp
                saludo={saludo}
                subtitulo={subTitulo}
            />
        );
        const textParrafo = wrapper.find('p').text();

        expect(textParrafo).toBe(subTitulo);
    });


});
