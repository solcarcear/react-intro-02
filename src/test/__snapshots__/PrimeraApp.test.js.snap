// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en primera app should return <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
    Hola soy carlos
  </h1>
  <p>
    Esta es mi primera vez
  </p>
</Fragment>
`;
